nbs/index.ipynb

```diff
@@ -1,23 +1,52 @@
 {
```

Comment:
<p>add usecases with rewrite_iter as well as usecases with imperative side effects for both rewrite and rewrite iter</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/DeanLight/graph_rewrite/pull/17/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='20'/>